### ISSUE TITLE
XHR: Calculate `total` of the response body if no `Content-Length` header was provided

### ISSUE THREAD

**Issue by kettanaito** [mswjs/interceptors#614]
Currently, the `total` of response events will be `0` if no explicit `Content-Length` response header was provided. We can improve that by inferring the total response body length from the actual response body. 

https://github.com/mswjs/interceptors/blob/e937d4fe1b8af54a8c87ffa508136cc126a0d832/src/interceptors/XMLHttpRequest/XMLHttpRequestController.ts#L306-L311

Something like this should suffice:

```js
const buffer = await response.clone().arrayBuffer()
const totalLength = buffer.byteLength
```

> Careful on cloning the response at the right time! 

This also needs in-browser tests.

**Comment by abhishekpaturkar** [comment]
Can you assign it to me.

**Comment by kettanaito** [comment]
## Released: v0.34.3 🎉

This has been released in v0.34.3!

- 📄 [**Release notes**](https://github.com/mswjs/interceptors/releases/tag/v0.34.3)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/interceptors/v/0.34.3)

Make sure to always update to the latest version (`npm i @mswjs/interceptors@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
